src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -526,13 +527,8 @@ protected override AuthenticationTicket AuthenticateCore()
         /// <param name="nonce">the nonce to remember.</param>
         /// <remarks><see cref="HttpResponse.Cookies.Append"/>is called to add a cookie with the name: 'OpenIdConnectAuthenticationDefaults.Nonce + <see cref="OpenIdConnectAuthenticationOptions.StringDataFormat.Protect"/>(nonce)'.
         /// The value of the cookie is: "N".</remarks>
-        private void WriteNonceCookie(string nonce)
+        private void WriteNonceCookie([NotNull] string nonce)
```

Comment:
@hishamco looks like you haven't reverted changes to this file.

Would be nice to switch restored line 531 from `IsNullOrWhiteSpace()` to `IsNullOrEmpty()` and use `nameof` on line 533 at this time.

At some point should also switch from `ArgumentNullException` to `ArgumentException` since an empty `string` isn't `null`. A few projects in a few repos have an `ArgumentCannotBeNullOrEmpty` resource for this case. Should add one (with matching wording) to this project. Again, not necessarily part of this PR.